fabric-tool-attributes-v1/src/main/java/net/fabricmc/fabric/api/tools/v1/ToolActor.java

```diff
@@ -0,0 +1,17 @@
+package net.fabricmc.fabric.api.tools.v1;
+
+import com.sun.istack.internal.Nullable;
+
+/**
+ * Interface for the user of a tool.
+ * @param <T> The type of actor that will use this tool.
+ */
+public interface ToolActor<T> {
+	ToolActor NO_ACTOR = () -> null;
```

Comment:
generics.

Reply:
Not sure how to do generics if there's a null.

Reply:
`java.lang.Void` like in `CompletableFuture<Void>`. I suggest moving this field to another package private class and use a method to get the dummy actor instead (like java `Collections.emptySet()` (type safe) vs `Collections.EMPTY_SET` (not type safe))